File: src/test/java/org/b0n541/pmcts/mcts/TreeTest.java
package org.b0n541.pmcts.mcts;

import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class TreeTest {
    @Test
    public void emptyTree() {
        final Tree tree = new Tree(new TestGameState(List.of("HEADS", "TAILS")));

        assertThat(tree.getSize()).isEqualTo(1);

        tree.getRootNode().expandPossibleMoves();

        assertThat(tree.getSize()).isEqualTo(3);

        tree.getRootNode().getChildren().get(0).expandPossibleMoves();

        assertThat(tree.getSize()).isEqualTo(5);

        tree.getRootNode().getChildren().get(1).expandPossibleMoves();

        assertThat(tree.getSize()).isEqualTo(7);
    }
}

File: src/test/kotlin/net/b0n541/ai/mcts/TreeTest.kt
package net.b0n541.ai.mcts

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class TreeTest {
    @Test
    fun emptyTree() {
        val tree = Tree(TestGameState(listOf("HEADS", "TAILS")))
        Assertions.assertThat(tree.size).isEqualTo(1)
        tree.rootNode.expandPossibleMoves()
        Assertions.assertThat(tree.size).isEqualTo(3)
        tree.rootNode.children()[0].expandPossibleMoves()
        Assertions.assertThat(tree.size).isEqualTo(5)
        tree.rootNode.children()[1].expandPossibleMoves()
        Assertions.assertThat(tree.size).isEqualTo(7)
    }
}